extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/Types.java

```diff
@@ -180,6 +180,18 @@ static boolean isAssignableFrom(DotName class1, DotName class2, IndexView index,
         return assignableInfo.isAssignableFrom(class2);
     }
 
+    static void indexHierarchy(ClassInfo classInfo, IndexView index) {
```

Comment:
Shouldn't this method return something?

Reply:
Not really. We just need to make sure all interfaces and subclasses are part of the bean archive index...

Reply:
Ah okay. I didn't know that you could add interfaces and subclasses to the bean archive by calli g `index.getClassByName`

Reply:
TBH, I don't understand either how this is supposed to work? Does it really index something when calling a method of an `IndexView`?

Reply:
So the bean archive index is "computing": https://github.com/quarkusio/quarkus/blob/main/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanArchives.java#L114-L120, i.e. something like [`CombinedIndexBuildItem#getComputingIndex()`](https://github.com/quarkusio/quarkus/blob/main/core/deployment/src/main/java/io/quarkus/deployment/builditem/CombinedIndexBuildItem.java#L39) that was probably inspired by the bean archive functionality. The problem is that methods like `getKnownImplementors()` do not compute/index additional classes.

Reply:
Looking at the sources, I believe it's because the `IndexView` implementation may be a `io.quarkus.arc.processor.BeanArchives.IndexWrapper` class, which adds some additional classes to the other method calls, but I may have misinterpreted that :)

Reply:
You're correct and this particular `IndexView` IS an instance of `BeanArchives.IndexWrapper`.